praw/models/reddit/subreddit.py

```diff
@@ -484,11 +485,6 @@ def _submit_media(self, data, timeout, without_websockets):
         # websocket creation happen right after the POST request,
         # otherwise you will have trouble.
 
-        if not isinstance(response, dict):
```

Comment:
Does this no longer occur?

Reply:
That check was there because in some cases the Reddit API was successfully making a malformed post. So, sometimes `response` would end up being a `Submission` object.

The issue was that, when submitting a video as an image (the file was actually a video, but the API was being told it was an image), the post would be created, but with a link to the temp S3 bucket instead of something on v.redd.it or i.redd.it. So this request would return a `Submission` object when the code here expects a dict. In contrast, trying to submit an image as a video would cause the API to respond with an error.

This code was here because I noticed that sometimes the request would go wrong in this way. By catching wrong filetypes *before* making network requests, I aim to avoid the case where the response is a `Submission`, thus making this section unnecessary.

Perhaps such a case is still achievable, such as if there is a video file with the incorrect extension `.jpg`. Since we determine filetype by extension, this might slip by our detection. I don't know whether the S3 bucket would complain upon upload and I can't test right now, but I will in the future.

If it turns out that I can cause the API to respond with a submission (that gets objected into a `Submission`), then I will put this code back in and craft a test to check for it.